Robinhood/Robinhood.py

```diff
@@ -174,81 +218,94 @@ def investment_profile(self):
         """Fetch investment_profile """
 
         res = self.session.get(self.endpoints['investment_profile'])
-        res.raise_for_status()  #will throw without auth
+        self.raiseForStatus(res)
         data = res.json()
 
         return data
 
+    def instruments(self, query=None, symbol=None, instrument=None):
+        """fetch instruments endpoint
 
-    def instruments(self, stock):
-        """Fetch instruments endpoint
+        Args:
+            query (str): search for ticker, e.g. by company name
+            symbol (str): find instrument by it's symbol
+            instrument (dict|str): valid instrument representation
 
-            Args:
-                stock (str): stock ticker
+        Returns:
+            (:obj:`dict`): JSON contents from `instruments` endpoint
 
             Returns:
                 (:obj:`dict`): JSON contents from `instruments` endpoint
         """
-
-        res = self.session.get(self.endpoints['instruments'], params={'query': stock.upper()})
-        res.raise_for_status()
+        res = None
+        if instrument:
```

Comment:
why did this get blown out?  This will require more tests to hit each if/else phrase

Reply:
Previously instruments only searched by a query string.  It was not using the symbol or instrument keywords available to be much more selective in searching.  This is one of the reason a number of the tests were failing for the 'MU' lookup as mentioned in #63 fixes.  

fwiw, I think I'm just going to merge that into this PR branch and make some of the recommended changes here to reduce the confusion..